modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/saml/SAMLErrorResponseTestCase.java

```diff
@@ -0,0 +1,307 @@
+                configContext);
+        ssoConfigServiceClient = new SAMLSSOConfigServiceClient(backendURL, sessionCookie);
+        httpClient = HttpClientBuilder.create().build();
+        createApplication();
+
+        // Starting tomcat
+        log.info("Starting Tomcat");
+        tomcatServer = getTomcat();
+
+        URL resourceUrl = getClass()
+                .getResource(File.separator + "samples" + File.separator + ARTIFACT_ID + ".war");
+        startTomcat(tomcatServer, "/" + ARTIFACT_ID, resourceUrl.getPath());
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void testClear() throws Exception {
+
+        deleteApplication();
+
+        ssoConfigServiceClient = null;
+        applicationManagementServiceClient = null;
+        httpClient = null;
+        //Stopping tomcat
+        tomcatServer.stop();
+        tomcatServer.destroy();
+        Thread.sleep(10000);
+    }
+
+    /**
+     * Tests whether the RelayState and ACS Url parameters are sent to the notification.do page in SAML error
+     * scenarios.
+     *
+     * @throws Exception
+     */
+    @Test(alwaysRun = true, description = "Testing Relay state and ACS parameters in SAML error response", groups =
+            "wso2.is")
+    public void testRelayStateAndACSWithSAMLErrorResponse() throws Exception {
+
+        // Create service provider config with mis-matching ACS to generate SAML error response
+        boolean isAddSuccess = ssoConfigServiceClient
+                .addServiceProvider(createSsoServiceProviderDTO());
+        assertTrue(isAddSuccess, "Adding a service provider has failed for " + ARTIFACT_ID);
+
+        HttpResponse response;
+        response = sendGetRequest(String.format(SAML_SSO_LOGIN_URL, ARTIFACT_ID, HTTP_POST_BINDING));
+        String samlRequest = extractDataFromResponse(response, "SAMLRequest", 5);
+        Map<String, String> paramters = new HashMap<String, String>();
+        paramters.put("SAMLRequest", samlRequest);
+        paramters.put("RelayState", RELAY_STATE);
+        response = sendSAMLMessage(SAML_SSO_URL, paramters);
+        EntityUtils.consume(response.getEntity());
+        response = sendRedirectRequest(response);
+
+        String sessionKey = extractDataFromResponse(response, "name=\"sessionDataKey\"", 1);
+        response = sendPOSTMessage(sessionKey);
+        EntityUtils.consume(response.getEntity());
+
+        String location = response.getFirstHeader("Location").getValue();
+        assertTrue(location.contains("RelayState=" + RELAY_STATE), "Redirection header to notification.do" +
+                " page should contain RelayState query param sent with the request");
+        assertTrue(location.contains("ACSUrl=" + URLEncoder.encode(String.format(ACS_URL,
+                ARTIFACT_ID), "UTF-8")), "Redirection header to notification.do" +
+                " page should contain ACS query param");
+    }
+
```

Comment:
In all the other cases, one new line between two methods. Here, there is an Unwanted new line. Any reason for that

Reply:
Fixed with #2238